atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/SweepableCellsWriter.java

```diff
@@ -84,15 +84,19 @@ private void insert(PartitionInfo info, WriteReference writeRef, boolean dedicat
                 .dedicatedRowNumber(dedicatedRow)
                 .build();
 
-        return SweepableCellsTable.SweepableCellsRow.of(
-                SweepQueueUtils.tsPartitionFine(info.timestamp()), metadata.persistToBytes());
+        return SweepableCellsTable.SweepableCellsRow.of(getTimestampOrPartition(info, dedicate),
+                metadata.persistToBytes());
     }
 
     private SweepableCellsTable.SweepableCellsColumnValue createColVal(long ts, long index, WriteReference writeRef) {
         SweepableCellsTable.SweepableCellsColumn col = SweepableCellsTable.SweepableCellsColumn.of(tsMod(ts), index);
         return SweepableCellsTable.SweepableCellsColumnValue.of(col, writeRef);
     }
 
+    private long getTimestampOrPartition(PartitionInfo info, boolean dedicate) {
+        return dedicate ? info.timestamp() : SweepQueueUtils.tsPartitionFine(info.timestamp());
+    }
+
```

Comment:
💯 yep, this version looks right.
We should probably have a test though given this passed multiple reviewers!

Reply:
Yes, but writing a test is actually nontrivial. One thing the old version would fail and the new would pass is a check that the timestamp modulus part of the column name must be equal for all cells in the row